build-tools/packages/build-cli/src/commands/release/prepare.ts

```diff
@@ -0,0 +1,266 @@
+		return {
+			result: true,
+		};
+	}
+
+	return {
+		result: false,
+		message: `Prerelease dependencies found. These release groups and packages must be released and integrated before a release can be done:\n\n${[
+			...releaseGroups.keys(),
+		].join("\n")}\n${[...packages.keys()].join("\n")}`,
+		fixCommand: "pnpm flub bump deps <release group or package>",
+	};
+}
+
+/**
+ * Returns true if there are no policy-check failures.
+ */
+async function checkPolicy(context: Context): Promise<CheckResult> {
+	// policy-check is scoped to the path that it's run in. Since we have multiple folders at the root that represent
+	// the client release group, we can't easily scope it to just the client. Thus, we always run it at the root just
+	// like we do in CI.
+	const result = await execa.command(`npm run policy-check`, {
+		cwd: context.gitRepo.resolvedRoot,
+	});
+
+	if (result.exitCode !== 0) {
+		return {
+			result: false,
+			message: "Policy check failed. These failures must be fixed before release.",
+			fixCommand: "pnpm run policy-check:fix",
+		};
+	}
+
+	return { result: true };
+}
+
+/**
+ * Returns true if there are no asserts to be tagged.
+ */
+async function checkAsserts(context: Context): Promise<CheckResult> {
+	// policy-check is scoped to the path that it's run in. Since we have multiple folders at the root that represent
+	// the client release group, we can't easily scope it to just the client. Thus, we always run it at the root just
+	// like we do in CI.
+	await execa.command(`npm run policy-check:asserts`, {
+		cwd: context.gitRepo.resolvedRoot,
+	});
+
+	// check for policy check violation
+	const afterPolicyCheckStatus = await context.gitRepo.getStatus();
+	if (afterPolicyCheckStatus !== "") {
+		const repo = new Repository({ baseDir: context.repo.resolvedRoot });
+		await repo.gitClient.reset(ResetMode.HARD);
+		return {
+			result: false,
+			message: "Found some untagged asserts. These should be tagged before release.",
+			fixCommand: "pnpm run policy-check:asserts",
+		};
+	}
+
+	return { result: true };
+}
+
+/**
+ * All the checks that should be executed. The checks are executed serially in this order.
+ */
+const allChecks: ReadonlyMap<string, CheckFunction> = new Map([
+	["Branch has no local changes", checkNoLocalChanges],
+	["Has a microsoft/FluidFramework remote", checkHasRemote],
+	["The local branch is up to date with the remote", checkBranchUpToDate],
+	["Dependencies are installed locally", checkDepsInstalled],
+	["Has no pre-release Fluid dependencies", checkHasNoPrereleaseDependencies],
+	["No repo policy violations", checkPolicy],
+	["No untagged asserts", checkAsserts],
+]);
+
+export class ReleasePrepareCommand extends BaseCommand<typeof ReleasePrepareCommand> {
+	static readonly summary =
+		`Runs checks on a local branch to verify it is ready to serve as the base for a release branch.`;
+
+	static readonly description = `Runs the following checks:\n\n- ${[...allChecks.keys()].join(
+		"\n- ",
+	)}`;
```

Comment:
Not sure if its better, but I think this also works and avoids duplciating the item prefix:

```suggestion
	static readonly description = `Runs the following checks:\n${["", ...allChecks.keys()].join(
		"\n- ",
	)}`;
```